src/Dependencies/PooledObjects/ArrayBuilder.cs

```diff
@@ -65,6 +74,22 @@ private ArrayBuilder(ObjectPool<ArrayBuilder<T>> pool)
             _pool = pool;
         }
 
+        [Conditional("DEBUG")]
+        public void Freeze()
+        {
+#if DEBUG
+            _frozen = true;
+#endif
+        }
+
+        [Conditional("DEBUG")]
+        public void AssertNotFrozen()
```

Comment:
Consider also calling in

- ToImmutableAndClear
- Count.set
- EnsureCapacity

Reply:
Done.  Whoops :(